### PR TITLE
added =c to test for chan equality in when clause

### PR DESCRIPTION
**Pull request by Ptival**
There are also syntactic changes and updates to all the necessary tests.

### CODE REVIEW

File: compiler/src/lex.mll [hunk 1]
```diff
@@ -34,8 +34,10 @@ rule token = parse
   | "ImmAfter" { IMMAFTER }
   | "ImmBefore" { IMMBEFORE }
   | "Match" { MATCH }
-  | "==" { EQUALITY }
+  | "=c" { EQC }
+  | "=n" { EQN }
   | "=" { EQ }
+  | "==" { EQUALITY }
```
Comment (ztatlock) [inline compiler/src/lex.mll]: Do we still need == ?

Comment (ztatlock) [inline compiler/src/lex.mll]: Did you get an email for the previous note or should I always @Ptival mention on these?

Comment (ztatlock) [inline compiler/src/lex.mll]: Sorry for spam if it already emailed you (and for this one).

Comment (Ptival) [inline compiler/src/lex.mll]: Yes I got all 3 emails. == is indeed obsolete.

**Comment by ztatlock** [comment]
Looks great! Merge at will.

**Pull request merged**